### DISCUSSION TITLE
Adguard Home on OSX doesn't respond to DNS requests from different subnets

### DISCUSSION THREAD

**Discussion by ely105** [AdguardTeam/AdGuardHome#6107] [Q&A]
I've noticed this issue for some time now and I haven't been able to solve it, only work around.  It appears to be MacOS specific.  But for some reason I can't get Adguard Home to respond to clients, ie DIG, etc. when the Adguard Home server is configured on a different subnet.  I've checked all firewall rules, turned off the firewall on MacOS, but still no success.  If i configure VLAN interfaces on the Mac and have Adguard "sitting" on each vlan with it's own IP on that subnet, it works correctly.  It just can't seem to respond to requests from different subnets.  

What I have tried/noticed:
 - This issue doesn't occur with an Ubuntu 22.04 installation of AGH sitting on the same networks - this works fine
 - I can ping from the clients to the Mac OS Adguard Home ip address and it responds correctly.  dig @ command just times out
 - no firewall rules exist between the subnets to interfere with operation, and again the ubuntu install works without issue
 - This has performed this way since OSX Monterey and i'm up to Ventura 13.5
 - Ive installed this on both x86 and M series mac minis with the same behavior

I've been using AGH on MacOS for some time and have just gone to the trouble of creating multiple vlan interfaces to get it to work.  but this is not always practical for my needs.  Hoping that there might be an easy config or fix out there that I haven't tried yet.  I'm running 2 instances of AGH on the networks so that there is some redundancy, which is why i have one running on ubuntu and one on a mac.

I thought i'd post as a question as I don't know if it's an actual bug.  

thanks!

**Comment by tug-benson** [comment]
Hello, 
Osx by default respond with ipv6 dns over public
```
scutil --dns | grep 'nameserver[[0-9]*]'
  nameserver[0] : 2a02:842b:fa7f:dc01:b69d:fdff:fe17:fb91
```
Set it to Link-local only, which sets the computer to only use IPv6 within your home, not on the internet. Some MacOS versions may even simply allow you to select Off as an option. See below to use Terminal to set IPv6 off if the option is not on the drop-down menu.

<img width="612" alt="image" src="https://github.com/user-attachments/assets/e4a2adb3-066d-4b87-9eab-b3592916dd6a" />

sudo networksetup -setv6off Ethernet
sudo networksetup -setv6off Wi-Fi

Cheers
TB
